lib/rules/forbid-foreign-prop-types.js

```diff
@@ -34,26 +48,58 @@ module.exports = {
       return node.parent.type === 'AssignmentExpression' && node.parent.left === node;
     }
 
+    function isInPropTypesAssignment(node) {
+      if (!allowInPropTypes) {
+        return false;
+      }
+
+      if (node.parent.parent.parent.parent.type === 'AssignmentExpression') {
+        const assignmentExpression = node.parent.parent.parent.parent;
+
+        if (
+          assignmentExpression.left &&
+          assignmentExpression.left.property &&
+          assignmentExpression.left.property.name === 'propTypes'
+        ) {
+          return true;
+        }
+      }
+      return false;
+    }
+
     return {
       MemberExpression: function(node) {
-        if (!node.computed && node.property && node.property.type === 'Identifier' &&
-            node.property.name === 'propTypes' && !isLeftSideOfAssignment(node) ||
-            node.property && node.property.type === 'Literal' &&
-            node.property.value === 'propTypes' && !isLeftSideOfAssignment(node)) {
+        if (
+          (!node.computed &&
+            node.property &&
+            node.property.type === 'Identifier' &&
+            node.property.name === 'propTypes' &&
+            !isLeftSideOfAssignment(node) &&
+            !isInPropTypesAssignment(node)) ||
+          (node.property &&
+            node.property.type === 'Literal' &&
+            node.property.value === 'propTypes' &&
+            !isLeftSideOfAssignment(node) &&
+            !isInPropTypesAssignment(node))
+        ) {
           context.report({
             node: node.property,
-            message: 'Using another component\'s propTypes is forbidden'
+            message:
+              'Using propTypes from another component is not safe because they may be removed in production builds'
```

Comment:
this is a super weird line break; can this all go on one line? (line length limits aren't important)

Reply:
This would have been done by Prettier. I'll revert it.